scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/constraints/activities/ActivityCreationTemplate.java

```diff
@@ -482,5 +484,18 @@ private  Optional<SchedulingActivityDirective> rootFindingHelper(
     return Optional.empty();
   }
 
+  private SimulationResults getLatestSimulationResults(final SimulationFacade facade, final Duration until){
+    final var latestConstraintsSimulationResults = facade.getLatestConstraintSimulationResults();
+    if(latestConstraintsSimulationResults.isEmpty()){
+      try {
+        facade.computeSimulationResultsUntil(until);
+        return facade.getLatestConstraintSimulationResults().get();
+      } catch(SimulationFacade.SimulationException e){
+        throw new RuntimeException("Simulation is in irreparable state");
```

Comment:
Because the only time we can do something about it is when we insert a new activity in simulation. This is done at line 294 with `removeAndInsertActivitiesFromSimulation` and you can see that we catch the simulation exception because we can actually remove the activity. In other cases, we use the results to instantiate a new instance (that is not yet inserted in the plan). If it fails, it means the error happened somewhere else, that an activity should not have been inserted and we cannot do anything about it other than scream. Basically, if this happens, it means that there is something wrong with the scheduler logic.